common/persistence/visibilityInterfaces.go

```diff
@@ -36,6 +36,7 @@ type (
 		Execution        s.WorkflowExecution
 		WorkflowTypeName string
 		StartTimestamp   int64
+		RetentionSeconds int64
```

Comment:
see comment above about naming. Can you take the workflowTimeout as input instead of the timeout+ttlBuffer ? The TTLBuffer can be hardcoded in the persistence file, I don't see that being changed frequently (because it is tied to the schema gc_grace_seconds anyways)

Reply:
True, it won't be changed frequently. Want to call out it's not tied to gc_grace_seconds. I decided to keep them the same here but they are not dependent. The TTL buffer was more me being extra safe for eg. if workflow timed out but network delays etc. take a little bit of extra time.